### PR TITLE
Add redirection to category pagination

### PR DESCRIPTION
**Pull request by okkez**

### CODE REVIEW

File: test/factories/site.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :site do
+    name "daimon-news"
+    fqdn "www.example.com"
```
Comment (tricknotes) [inline test/factories/site.rb]: `Site#fqdn` should be a uniqueness attribute.
So should we make it also uniqueness in test data?

File: test/routes/category_route_test.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class CategoryRouteTest < ActionDispatch::IntegrationTest
+  def setup
+    @site = create(:site)
+    @category = create(:category, site_id: @site.id)
```
Comment (tricknotes) [inline test/routes/category_route_test.rb]: Should it be type safe?

``` ruby
create(:category, site: @site)
```

File: test/factories/category.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+  factory :category do
+    name "category1"
+    description "category1 description\n"
+    slug "category1"
+    order 1
```
Comment (tricknotes) [inline test/factories/category.rb]: Could we use `sequence` ?

**Comment by tricknotes** [comment]
I left some comments however this PR is reasonable :+1:

File: test/factories/site.rb (hunk 1 above)
Comment (okkez) [inline test/factories/site.rb]: I think so.
But I cannot set `request.server` in test...

Comment (okkez) [inline test/factories/site.rb]: Done!

File: config/routes.rb [hunk 4]
```diff
@@ -24,6 +24,7 @@
   concern :site do
     resources :posts, only: %i(show), constraints: {id: /\d+/}, path: '/'
```
Comment (mtsmfm) [inline config/routes.rb:25]: By the way, we should compatible with old URL for Posts#show.
Could you add this line?

```
get '/:id/:page' => redirect('/%{id}?page=%{page}'), constraints: {id: /\d+/, page: /\d+/} # To compatible with old URL
```

**Comment by mtsmfm** [comment]
:cool:

File: config/routes.rb (hunk 4 above)
Comment (okkez) [inline config/routes.rb:25]: OK, I will do it

File: config/routes.rb [hunk 5]
```diff
@@ -25,6 +25,10 @@
...
     resources :categories, only: %i(show), path: 'category'
 
+    # compatibility for old URL
+    get ':id/page/:page' => redirect('%{id}?page=%{page}'), constraints: {original_id: /\d+/, page: /\d+/}
```
Comment (mtsmfm) [inline config/routes.rb]: `:id/:page` is correct.

WordPress doesn't have `/page/` for post...

Comment (okkez) [inline config/routes.rb]: Sorry, I've fixed.

**Comment by mtsmfm** [comment]
:shipit:

**Pull request merged**